### PR TITLE
Add missing APIs for multi-asset

### PR DESCRIPTION
**Pull request by rvl**
## Issue Number

ADP-698

## Overview

- Adds assets to the responses of the coin selections endpoints.
- Adds assets to the `/byron-wallets` endpoints.

### CODE REVIEW

**Comment by KtorZ** [comment]
@rvl shall we also tackle this quick one at the same time: https://github.com/input-output-hk/cardano-wallet/pull/2476#issuecomment-769862468

**Comment by rvl** [comment]
Yes, coming up...

File: lib/core/src/Cardano/Wallet/Primitive/Types/TokenMap.hs [hunk 1]
```diff
@@ -213,6 +215,8 @@ instance PartialOrd TokenMap where
         (getAssets m1 `Set.union` getAssets m2)
 
 instance NFData TokenMap
+instance Hashable TokenMap where
+    hashWithSalt = hashUsing (Map.toList . unTokenMap)
```
Comment (KtorZ) [inline lib/core/src/Cardano/Wallet/Primitive/Types/TokenMap.hs:219]: Can't `GeneralizedNewtypeDeriving` or `DerivingVia` handle that just fine :thinking: ?

File: specifications/api/swagger.yaml [hunk 2]
```diff
@@ -996,6 +996,7 @@ x-transactionResolvedInputs: &transactionResolvedInputs
     properties:
       address: *addressId
       amount: *amount
+      assets: *walletAssets
```
Comment (KtorZ) [inline specifications/api/swagger.yaml:1000]: :+1:

**Review by KtorZ — APPROVED** [review]
:ok_hand:

File: lib/core/src/Cardano/Wallet/Primitive/Types/TokenMap.hs (hunk 1 above)
Comment (rvl) [reply lib/core/src/Cardano/Wallet/Primitive/Types/TokenMap.hs:219]: It didn't seem to work - I think perhaps Map didn't have a Hashable instance.

File: lib/core/src/Cardano/Wallet/Api.hs [hunk 3]
```diff
@@ -616,6 +622,38 @@ type PutByronWalletPassphrase = "byron-wallets"
...
+    :> Capture "policyId" (ApiT TokenPolicyId)
+    :> Capture "assetName" (ApiT TokenName)
+    :> Get '[JSON] ApiAsset
+
+-- | https://input-output-hk.github.io/cardano-wallet/api/#operation/getByronAssetDefault
+type GetByronAssetDefault = "byron-wallets"
+    :> Capture "walletId" (ApiT WalletId)
+    :> "assets"
+    :> Capture "policyId" (ApiT TokenPolicyId)
+    :> Get '[JSON] ApiAsset
```
Comment (KtorZ) [inline lib/core/src/Cardano/Wallet/Api.hs:655]: :+1:

**Comment by piotr-iohk** [comment]
As far as I see assets are now present on `inputs` and `outputs` on the coin selection response, but they are not on `change`. Should there be on change too @rvl?

Exemplary response:

```
{
  "inputs": [
    {
      "amount": {
        "quantity": 353684111,
        "unit": "lovelace"
      },
      "address": "addr_test1qzvw02vau2jf8vmwav657w5j9grstsx3sp4cgnqh9agnz48l0dw5r75vk42mv3ykq8vyjeaanvpytg79xqzymqy5acmqy0h2p9",
      "id": "b2c4b2725a1cbe6a46b843368f525b87d3a4e1388834a18b39f2d0db7d5c069e",
      "derivation_path": [
        "1852H",
        "1815H",
        "0H",
        "1",
        "287"
      ],
      "assets": [
        {
          "asset_name": "6861707079636f696e",
          "quantity": 4000,
          "policy_id": "789ef8ae89617f34c07f7f6a12e4d65146f958c0bc15a97b4ff169f1"
        },
        {
          "asset_name": "7375706572636f696e",
          "quantity": 4000,
          "policy_id": "789ef8ae89617f34c07f7f6a12e4d65146f958c0bc15a97b4ff169f1"
        }
      ],
      "index": 3
    }
  ],
  "deposits": [],
  "change": [
    {
      "amount": {
        "quantity": 351505014,
        "unit": "lovelace"
      },
      "address": "addr_test1qqes885m0524cgxk84awhlhp38qf9q2p9c6t99zs0smdy3ll0dw5r75vk42mv3ykq8vyjeaanvpytg79xqzymqy5acmqx4x7rl",
      "derivation_path": [
        "1852H",
        "1815H",
        "0H",
        "1",
        "288"
      ]
    }
  ],
  "outputs": [
    {
      "amount": {
        "quantity": 2000000,
        "unit": "lovelace"
      },
      "address": "37btjrVyb4KF6KtvXYUxDHqjAwfKAuDzP71UEuG6vh87rjstvB8o1Muu6KaeGAF8xZro63Zc5VAYNnQtcci4vT8jhLuNtwku9x1YgS18LHFAhxxTi8",
      "assets": [
        {
          "asset_name": "6861707079636f696e",
          "quantity": 1000,
          "policy_id": "789ef8ae89617f34c07f7f6a12e4d65146f958c0bc15a97b4ff169f1"
        },
        {
          "asset_name": "7375706572636f696e",
          "quantity": 1000,
          "policy_id": "789ef8ae89617f34c07f7f6a12e4d65146f958c0bc15a97b4ff169f1"
        }
      ]
    }
  ]
}
```

**Comment by rvl** [comment]
Ah yes ... fixed now. I think that's the last one.

ApiTransaction doesn't have an assets field corresponding to the amount field, but that's a known limitation (ADP-683).

**Comment by rvl** [comment]
bors retry

**Comment by KtorZ** [comment]
bors retry

**Pull request merged**